### DISCUSSION TITLE
Welcome to oce Discussions!

### DISCUSSION THREAD

**Discussion by dankelley** [dankelley/oce#1754] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by dankelley** [comment]
I'm Dan Kelley, a Physical Oceanographer at Dalhousie university and a co-author of the oce package.  (Shameless plug: I also have a book that might interest oceanographers: Kelley, Dan E. Oceanographic Analysis with R. New York: Springer-Verlag, 2018. https://www.springer.com/us/book/9781493988426.)

I do the majority of oce coding, and I monitor the issues regularly, although on the latter I probably am not as helpful as @richardsc.

**Comment by coffm049-2** [comment]
Hi there! I'm newer to using CTD-esque data. I'm working with some data that reports the variable "specc" which corresponds to the long name "Specific Conductance". 

Using the oce package I see that this column automatically gets mapped to "Conductance" (as reported in line 156 of [this file](https://github.com/dankelley/oce/blob/692ff2a31499ce5b12c321c86aacaf5689a2f20e/man/cnvName2oceName.Rd#L156).

Is there any easy way for me to solve this? I was planning on forking and using my own version with a very slight change to account for this.

Thank you,
Christian

**Reply by dankelley** [reply]
If you're read it into an object named `d`, say, you can do as follows.
```R
dn <- names(d@data)
dn[dn == "oldname"] <- "newname"
names(d@data) <- dn
```
etc.

I wouldn't bother altering software and rebuilding because if you do that, then your R scripts will not be transportable.  Doing as above means you could send the code to someone else.

PS. I am actually a bit confused by your message, because the line of code to which you link is not naming it as you are suggesting.  But, anyway, the above is how you can rename things.  You can also insert information about the name change into the processing-log, if you wish.

**Reply by coffm049-2** [reply]
Thank you for your quick response. It is much appreciated! :)

Your suggested code does make sense. However, I will be processing hundreds to thousands of these files, some of which have "specc" and some which don't. So I need to programmatically decide if/when to make this naming substitution. Although I might be able to do this with a regex on the header from the raw text file (if it contains "specc" or not). 

> PS. I am actually a bit confused by your message, because the line of code to which you link is not naming it as you are suggesting. But, anyway, the above is how you can rename things. You can also insert information about the name change into the processing-log, if you wish.

I mightbe misunderstanding, however what by my estimation is going on is the following mapping "specc" -> "conductivity". However, this is undesirable for my case. I wish to have the mapping "specc" -> "specific conductance". The code that I would change would be lines 533-534 in [this file](https://github.com/dankelley/oce/blob/692ff2a31499ce5b12c321c86aacaf5689a2f20e/R/ctd.sbe.R#L136).

**Reply by richardsc** [reply]
I think this conversation would be more appropriate as an "Issue", i.e. here:

https://github.com/dankelley/oce/issues

@coffm049-2 would you mind opening an issue and moving the relevant parts of your request there? It would also help if you could provide some details about the instrument/file that you're working with (e.g. what do you mean by "CTD-esque"?).

**Reply by richardsc** [reply]
And to clarify, I think that the issue could be boiled down to:

"specc" is incorrectly converted to "conductivity", but should be "specificConductance"

**Reply by coffm049-2** [reply]
Yes no problem. I'll go ahead and transfer it over to Issues.

**Reply by dankelley** [reply]
If it's an error, we can change oce, of course.  I must have inferred that this was a type of conductivity from some manual.  Note that if it's wrong, then perhaps the unit is wrong also, and that would mean a few more lines to change.

PS. the code I sent will work whether the file has that variable or not, as shown below
``` r
a <- data.frame(a=1:3, b=10:12)
str(a)
#> 'data.frame':    3 obs. of  2 variables:
#>  $ a: int  1 2 3
#>  $ b: int  10 11 12
n <- names(a)
# This won't change anything...
n[n == "c"] <- "will not happen"
names(a) <- n
str(a)
#> 'data.frame':    3 obs. of  2 variables:
#>  $ a: int  1 2 3
#>  $ b: int  10 11 12

# ... but this does.
n[n == "b"] <- "new b"
names(a) <- n
str(a)
#> 'data.frame':    3 obs. of  2 variables:
#>  $ a    : int  1 2 3
#>  $ new b: int  10 11 12
```

<sup>Created on 2024-05-02 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

**Reply by dankelley** [reply]
I see that is *is* an error (see below from an SBE manual) and I'll make the fix once I see an issue.  I can also set up the issue myself but it's nice to see issues from users.

<img width="774" alt="Screenshot 2024-05-02 at 2 52 40 PM" src="https://github.com/dankelley/oce/assets/99469/fa5f1862-6c10-4c19-b34e-26c9ef29b4a5">

**Reply by coffm049-2** [reply]
Thank you for checking up on this! I've made the issue 

https://github.com/dankelley/oce/issues/2211#issue-2276154427

**Comment by rvaillancourt** [comment]
Hi everyone, I wonder if someone can help me. I am trying to calculate and display error for the mean values created using binAverage function in OCE. I want to calculate the standard deviation of each bin and then display it on an xy plot, preferably as shaded areas above and below the mean line.  Any ideas?

**Reply by richardsc** [reply]
Hi @rvaillancourt -- sorry for the slow reply on this. @dankelley and I don't always get notifications about the "Discussions" from Github for some reason.

If you don't mind, I'd like to move your question over to the "Issues" tab, to better keep track of it as a distinct topic. Things tend to get lost if they are just buried in this "Welcome to oce" discussion.

Thanks for using oce! 😄
